### DISCUSSION TITLE
recognizing labelled na values

### DISCUSSION THREAD

**Discussion by Cghlewis** [rstudio/pointblank#427] [Q&A]
Hi,

I am working with labelled na values, assigned through either `labelled::na_range()` or `labelled::na_values()`. And prior to now, I did not think pointblank would recognize those values as NA values when you use a validation function such as `col_vals_between` with the argument `na_pass = TRUE`. But now I am working with this more and it looks like pointblank may recognize `labelled` na values (like -99) to be NA in the `na_pass = TRUE` argument. Is this correct or am I misinterpreting this?

Thank you!

**Comment by rich-iannone** [comment]
Sorry for not getting to this sooner. I didn’t do any work to interpret labeled NA values. In fact I don’t know much about this but it is interesting. Would you be able to provide an MRE that demonstrates the odd behavior? Perhaps there is something that could be done better in the package.

**Reply by Cghlewis** [reply]
Of course. I'm on R version 4.1.2 and RStudio version 2021.09.0.

You'll see below that when I add NA value labels using the labelled package, those values are recognized as NA in both col_vals_between() and col_vals_in_set(), but not in col_vals_gte().
``` r
# load packages
library(tidyverse)
#> Warning: package 'ggplot2' was built under R version 4.1.3
#> Warning: package 'tibble' was built under R version 4.1.3
#> Warning: package 'readr' was built under R version 4.1.3
#> Warning: package 'dplyr' was built under R version 4.1.3
library(labelled)
#> Warning: package 'labelled' was built under R version 4.1.3
library(pointblank)
library(reprex)

# create data
data <- tibble::tribble(~id, ~score, ~var1,
                        10, 32, 2,
                        11, -99, 3,
                        12, 27, -99)
# review data
head(data)
#> # A tibble: 3 x 3
#>      id score  var1
#>   <dbl> <dbl> <dbl>
#> 1    10    32     2
#> 2    11   -99     3
#> 3    12    27   -99

# assign NA value labels
data_lab <- data %>%
  set_na_range(score = c(-99,-1), 
               var1 = c(-99, -1))

# review data
head(data_lab)
#> # A tibble: 3 x 3
#>      id     score      var1
#>   <dbl> <dbl+lbl> <dbl+lbl>
#> 1    10   32         2     
#> 2    11  -99 (NA)    3     
#> 3    12   27       -99 (NA)


# validate data
val_data <- create_agent(data_lab) %>%
  col_vals_between(columns = 
                     vars(score), 
                   left = 0, right = 50, na_pass = TRUE) %>%
  col_vals_in_set(columns = 
                    vars(var1), 
                  set = c(0, 1, 2, 3, 4, NA)) %>%
  col_vals_gte(columns = vars(score), value = 0, na_pass = TRUE) %>%
  interrogate()
```

<sup>Created on 2022-08-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
